### DISCUSSION TITLE
08.hmu.csv not loaded in local path

### DISCUSSION THREAD

**Discussion by avanmourik** [john30/ebusd#1009] [Q&A] [answered]
Being new to this community , I have been trying to get my ebusd up and running for the combination of the following equipment on a Raspberry Pi model 4 arm71 with ebusd 23.2:

scan.08  = Vaillant;HMU00;0607;5103 (VWL 75/6) heatpump
scan.15  = Vaillant;CTLV2;0514;1104 (VRC 720/2) thermostat
scan.76  = Vaillant;VWZIO;0202;0103 (VWZ AI) control unit.

Data are transferred via the USB connection with ID ttyACM0. 
In order not to make this report too long I have sticked to (in my view) relevant output only

I have noticed there are several issues around these components , which i also have, and part of them are unsolved, whereas others have a variety of underlying causes mentioned.
However there is one particular issue I do not understand at all :

Using the CONFIG string : EBUSD_OPTS="--scanconfig -d ens:/dev/ttyACM0 --latency=150  --configpath=http://cfg.ebusd.eu/" , the system components are all found and ebusctl info reports a.o. :

address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0607;HW=5103", loaded "vaillant/08.hmu.csv"
The ebusctl find command reports the usual datalist, with "no data stored", found in many other reported issues
address 03: master #11
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=CTLV2;SW=0514;HW=1104"
address 31: master #8, ebusd
address 36: slave #8, ebusd
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0202;HW=0103"
The 08.hmu.csv file is loaded !

 The /var/log/ebusd.log reports the following:

2023-09-09 18:37:05.975 [main notice] read common config file vaillant/broadcast.csv
2023-09-09 18:37:06.061 [main notice] read common config file vaillant/scan.csv
2023-09-09 18:37:06.148 [main notice] read common config file vaillant/general.csv

So the non-specific files are read as well

Further down:
--------------------------------------------------------------------------------------
[main notice] read scan config file vaillant/08.hmu.csv for ID "hmu00", SW0607, HW5103
2023-09-09 18:37:06.348 [update notice] received read hmu State QQ=71: 0;0;160;0
2023-09-09 18:37:06.421 [main notice] found messages: 67 (0 conditional on 0 conditions, 0 poll, 10 update)

Not sure where the 67 messages come from, the file is much shorter.

The csv's for scan 15 and scan 76 are not available , so no csv loaded for those particular pieces of equipment.

This seems logical behaviour (apart from the mostly zero output)

Now in order to try to play around with the csv files and some of the variants found on the internet, I changed the the configuration path to a loca path, which gives more control over the csv files:
EBUSD_OPTS="--scanconfig -d ens:/dev/ttyACM0 --latency=150  --configpath=/etc/ebusd/vaillant"

the /etc/ebusd/vaillant directory contained the following files (copies of the original website files in ebusd-2.1.x):

08.hmu.csv  broadcast.csv  general.csv  memory.csv  scan.csv  _templates.csv

The command $ ebusctl info now reports :

address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0607;HW=5103"
So it ignores the 08.hmu.csv file in the config path!

The /var/log/ebusd.log file reports:

2023-09-09 02:59:45.172 [bus notice] scan 08: ;Vaillant;HMU00;0607;5103
2023-09-09 02:59:45.172 [update notice] store 08 ident: done
2023-09-09 02:59:45.172 [update notice] sent scan-read scan.08  QQ=31: Vaillant;HMU00;0607;5103
2023-09-09 02:59:45.172 [bus notice] scan 08: ;Vaillant;HMU00;0607;5103
2023-09-09 02:59:45.173 [main error] unable to load scan config 08: list files in vaillant ERR: element not found
2023-09-09 02:59:45.173 [main error] scan config 08: ERR: element not found

It also seems to ignore scan, general and broadcast  csv-files completely (no sign of reading in the logfile .

To make this even more interesting , I added a 15.ctlv2.csv file to the /etc/ebusd/vaillant/ directory. This file was taken from :
https://raw.githubusercontent.com/janphkre/ebusd-configuration/feature/support-ctlv2/ebusd-2.1.x/en/vaillant/15.ctlv2.csv.

What happens next is, that the message titles in this file are partly read :
$ ebusctl find:
ctlv2 AdaptHeatCurve = no data stored
ctlv2 ContinousHeating = no data stored
ctlv2 CylinderChargeHyst = no data stored
ctlv2 CylinderChargeOffset = no data stored
ctlv2 Date = no data stored
ctlv2 FrostOverRideTime = no data stored
ctlv2 HwcLockTime = no data stored
ctlv2 HwcParallelLoading = off
ctlv2 HydraulicScheme = no data stored
ctlv2 MaintenanceDate = no data stored
ctlv2 MaxCylinderChargeTime = no data stored
ctlv2 MaxRoomHumidity = no data stored
ctlv2 PumpAdditionalTime = no data stored
ctlv2 Time = no data stored
ctlv2 WaterPressure = no data stored
memory eeprom = no data stored
memory ram = no data stored
scan.08  = Vaillant;HMU00;0607;5103
scan.15  = Vaillant;CTLV2;0514;1104
scan.76  = Vaillant;VWZIO;0202;0103

However the file is never loaded (visible in $ ebusctl info):
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0607;HW=5103" 
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=CTLV2;SW=0514;HW=1104" (>>> loaded message should appear here)
address 31: master #8, ebusd
address 36: slave #8, ebusd
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0202;HW=0103"

upon discovery of the CTLV2 /var/log/ebusd.log reports:

[bus notice] scan 15: ;Vaillant;CTLV2;0514;1104
2023-09-12 13:32:26.748 [update notice] store 15 ident: done
2023-09-12 13:32:26.748 [update notice] sent scan-read scan.15  QQ=31: Vaillant;CTLV2;0514;1104
2023-09-12 13:32:26.748 [bus notice] scan 15: ;Vaillant;CTLV2;0514;1104
2023-09-12 13:32:26.748 [main error] unable to load scan config 15: list files in vaillant ERR: element not found
2023-09-12 13:32:26.748 [main error] scan config 15: ERR: element not found.

Direct read requests for the ctlv2 values result in (example only, but true for all values)

ctlv2 MaintenanceDate = ERR: SYN received.

Hoping for some light in the dark.....

**Comment by edsub** [comment]
This looks quite similar to my situation.
If a csv file is not loaded while a component is recognized, the cause might be:
1. Rights on the csv file. I ended up in giving all csv files a 777.
2. Time. Sometimes it takes an awful long time to have ebusd connect the component identifiers to a csv.

I have the feeling the latter has some strange cause that should be fixed. E.g. I notice a significantly faster connection of csv to component when I start ebusd with --loglevel=debug .... This should NOT be required (and afaik ot is not, except with me....).

One of the effects (in my case) is that I cannot successfully do a read or a write on an identified fielld.

**Reply by avanmourik** [reply]
Tried both options
1) made all files in the directory rwx and verified it happened .
No change in behaviour (no binding of 08.hmu.csv to system component; same for 15.ctlv2.csv)
just saw that https://upd.ebusd.eu/ 08.hmu.csv has a new version (23/9) ; so need to check that one, however no clue how to get it in my local directory, since the downloadable version on https://github.com/john30/ebusd-configuration/blob/master/ebusd-2.1.x/en/vaillant/08.hmu.csv is 8 months old!
2) added the loglevel --loglevel=debug and this indeed confirmed your observation. Without introducing this option it ebusd revealed only 2 out of the 3 components in more than half an hour. After adding the loglevel option is found the 3 components in a few minutes.

**Comment by avanmourik** [comment, marked as answer]
problem solved.
setting logfile to debug revealed the problem :- errors were found in most csv-files
linenumber indicated where the errors were.
They were mostly related to lacking definitions in _templates.csv.
searched the web to add missing defs as good as possibe
then commented out the remaining error producing lines in the other csv files until no errors anymore.
Then the files were read and attached to components of my system.

RESULT:
ebusctl info
version: ebusd 23.2.23.2
update check: OK, device firmware 1[3911] available, memory.csv: different version available, vaillant/08.hmu.csv: different version available, vaillant/errors.inc: different version available, vaillant/general.csv: different version available
device: /dev/ttyACM0, enhanced high speed, firmware 1.1[3903].1[3903]
signal: acquired
symbol rate: 42
max symbol rate: 191
min arbitration micros: 5
max arbitration micros: 17
min symbol latency: 4
max symbol latency: 4
scan: finished
reconnects: 0
masters: 4
messages: 523
conditional: 0
poll: 0
update: 1
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0607;HW=5103", loaded "vaillant/08.hmu.csv"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=CTLV2;SW=0514;HW=1104", loaded "vaillant/15.ctlv2.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZIO;SW=0202;HW=0103", loaded "vaillant/76.vwzio.csv"
